documentation/modules/mtv-aio-buffer.adoc

```diff
@@ -0,0 +1,250 @@
+// Module included in the following assemblies:
+//
+// * documentation/doc-Migration_Toolkit_for_Virtualization/master.adoc
+
+:_content-type: CONCEPT
+[id="mtv-aio-buffer_{context}"]
+= Increasing asynchronous I/O (AIO) sizes and buffer counts for NBD transport mode
+
+This document describes how to change NBD transport NFC parameters for increased migration performance when using the {project-first} product.
+
+[WARNING]
+.Cold Migration Only
+====
+Using AIO buffering is only suitable for Cold Migration use cases.
+
+* Disable AIO settings before initializing Warm Migration. For more details, see xref:mtv-disable-aio-buffer_mtv[Disabling AIO Buffer Configuration].
+====
+
+.Additional information
+
+* link:https://docs.vmware.com/en/VMware-vSphere/7.0/vsphere-vddk-programming-guide/GUID-5D166ED1-7205-4110-8D72-0C51BB63CC3D.html[Best Practices for NBD Transport]
+* link:https://forums.veeam.com/vmware-vsphere-f24/solution-for-poor-nbd-performance-t93084.html[Solution for poor NBD performance]
+
+[id="mtv-aio-buffer-key-findings_{context}"]
+== Key findings
+
+* The best migration performance was achieved by migrating using multiple VMs (10) on a single ESXi host with the following values:
+** `VixDiskLib.nfcAio.Session.BufSizeIn64KB=16`
+** `vixDiskLib.nfcAio.Session.BufCount=4`
+
+* The following improvements were noted when using AIO buffer (Asynchronous Buffer Counts) settings:
+** Migration time was reduced by *31.1%*, from `0:24:32` to `0:16:54`.
+** Read rate was increased from `347.83MB/s` to `504.93MB/s`.
+
+* There was no significant improvement observed when using AIO buffer settings with a single VM.
+
+* There was no significant improvement observed when using AIO buffer settings with multiple VMs from multiple hosts.
+
+[id="mtv-enable-aio-buffer_{context}"]
+== Enabling AIO buffer configuration
+
+.Validating Controller Pod support for AIO values
+
+* Ensure that the `forklift-controller` pod in the `openshift-mtv` namespace supports the AIO buffer values.
++
+Since the pod name prefix is dynamic, check the pod name first by running the following command:
++
+[source,terminal]
+----
+oc get pods -n openshift-mtv | grep forklift-controller | awk '{print $1}'
+----
++
+The example output is as follows:
++
+[source,termina]
+----
+forklift-controller-667f57c8f8-qllnx
+----
++
+[NOTE]
+====
+This is the pod name prefix from the example: `forklift-controller-667f57c8f8-qllnx`
+====
+
+* Check the environment variables of the pod by running:
++
+[source,termina]
+----
+oc get pod forklift-controller-667f57c8f8-qllnx -n openshift-mtv -o yaml
+----
++
+* Check for the following lines in the output:
++
+[source,termina]
+----
+...
+\- name: VIRT\_V2V\_EXTRA\_ARGS
+\- name: VIRT\_V2V\_EXTRA\_CONF\_CONFIG\_MAP
+...
+----
+
+.Editing ForkliftController Configuration
+
+* In the `openshift-mtv namespace`, edit the `ForkliftController` object to include the AIO buffer values.
++
```

Comment:
add below line 84 , the command to edit the ForkliftController as  follow : 
run the following command to edit the ForkliftController object :
oc edit forkliftcontroller -n openshift-mtv